providers/go-feature-flag/pom.xml

```
@@ -50,5 +50,24 @@
             <version>4.11.0</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>32.1.1-jre</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>2.0.7</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j2-impl</artifactId>
+            <version>2.20.0</version>
```

I see that log4j is used only for the tests, so I will suggest to scope of this dependency to tests only.

```suggestion
            <version>2.20.0</version>
            <scope>test</test>
```

---

Used logging also at non-test GoFeatureFlagProvider, with Slf4j annotation and _log_ field.

---

Yes but your logger comes from slf4j not from log4j (except in the test).
In the provider itself the [dependency to `org.slf4j`](https://github.com/liran2000/java-sdk-contrib/blob/a4892214f3ca435593fb3bc4db38d0b63fcd62cf/providers/go-feature-flag/pom.xml#L60C9-L64) is enough, no need to specify an implementation.